### ISSUE TITLE
'Refresh page' menu item does not refresh the menu bar

### ISSUE THREAD

**Issue by RieksJ** [AmpersandTarski/Ampersand#434]
<img width="497" alt="picture1" src="https://cloud.githubusercontent.com/assets/8522589/15426306/f399057a-1e8d-11e6-9a68-9f0f88617b0b.png">

When you click on 'Refresh page' (see figure above), then the menu bar (which in the picture contains items 'Group Chats', 'My Profile' and 'Login') is not refreshed.

**Comment by Michiel-s** [comment]
That is intended behavior. Maybe it is better to call it 'Refresh interface (data)', as that is what it does.

**Comment by RieksJ** [comment]
Ok that it is intended behavior. Can we then change its intention so that it also refreshes the menu bar? Or does it make more sense to discuss 'refreshments' of page(part)s in a more general sense (enabeling e.g. rules (as used in Login) to force a refresh)?

**Comment by Michiel-s** [comment]
Here's a proposal:

**Use case:**
When `loginEnabled` is set to true in localSetting.php and the SIAM module is used, the following interface is used to determine the allowed roles (which can be selected in the role menu)
```
INTERFACE "SessionRoles" FOR SYSTEM : sessionAllowedRoles[SESSION*Role] cRud BOX [ "ignored": I]
```
You have another interface (e.g. Login page) to allow a user to login and let the ExecEngine (de)populate the sessionAllowedRoles relation accordingly. This change however is not reflected in the role menu because it requires a refresh of another part of the web application (in this case, the role menu).

Other use cases regarding session variables are also thinkable, e.g. the excel import is only visible for certain roles.

**Proposed solution**
In the return object of an interface post, put, patch or delete call (changes) we provide a flag that indicates whether or not the web app should refresh (get) the session variables. In the backend we keep already track of affected relations. Such 'refresh'-flag must be set when a session relation is affected AND the transaction is committed.

**Benefit for user**
No refresh of the page (F5) is required anymore after Login or any other change in an interface that affects session variables.

**Impact of change**
Haskell part: none
Database: none
Backend: implement 'refresh'-flag.
Frontend: 'refresh'-flag should trigger calls to get session variables

**Comment by RieksJ** [comment]
Let's do it. It would improve things, perhaps even solve the thing entirely.

Could we also get an ExecEngine function that sets the refresh-flag? While it is not needed to solve the login stuff etc., I showed you cases where changes in some BOX affected contents of other BOXes, which is not shown unless you refresh the page entirely. We could use this ExecEngine function to play with refresh, to see if/how we would want refreshes to be done more generally.

**Comment by hanjoosten** [comment]
Somewhat better now, but not totally working yet. As an example, I have the logon stuff in RAP:

 1. Open the Login interface
 2. check the checkbox 'please help'
 3. select any user. **Observation: The menu is refreshed. Very nice!**
 4. check the logout checkbox. We get the following screen:

![image](https://cloud.githubusercontent.com/assets/1159672/18990007/8351a142-870f-11e6-9364-7f04ca899e98.png)

 5. Click the Login menu-item. Observation: Nothing happens!
 6. Hit the F5 button: the screen is refreshed, enabling the user to log in.

**Comment by Michiel-s** [comment]
@hanjoosten this github issue is about the menubar refreshing or not. What you point out is the other part of the 'F5' issue, addressed by #412 and #476 

Please close this issue if the **navigation bar** is showing correct behavior.
